snuba/replacers/errors_replacer.py

```diff
@@ -263,6 +262,10 @@ def get_timestamp_condition(msg_value: Optional[str], operator: str) -> str:
         if not msg_value:
             return ""
 
+        if "+" in msg_value:  # e.g. "2023-08-28T03:05:38+00:00"
+            msg_value, _ = msg_value.rsplit("+", 1)
+            msg_value += ".000Z"
```

Comment:
how about:
```suggestion
            datetime.datetime.strftime(datetime.datetime.strptime(msg_value, other_format), settings.PAYLOAD_DATETIME_FORMAT)
```

where:
`other_format = "%Y-%m-%dT%H:%M:%S.%f%z"`

We do have users running this in dev, self hosted where it wont work if it's not UTC